### PR TITLE
complete files

### PR DESCRIPTION
**Pull request by emanuelmustea**

### CODE REVIEW

**Comment by DarianRusu** [comment]
sugerez sa pui folderul de /dist in .gitignore, nu prea au motiv sa fie pushed fisierele si pot cauza merge conflicts degeaba

File: src/carousel/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    this.carouselContainerWidth = this.carouselElement.offsetWidth;
+    this.setState({ leftPosition: -1 * this.carouselContainerWidth });
+  }
+  render() {
+    let { imgs } = this.props;
+    imgs = this.cloneImages(imgs);
+    return (
+      <div className="carousel" ref={ref => (this.carouselElement = ref)}>
+        {this.renderSliderElement(imgs)}
+        {this.renderButtons('prev', () => this.changePhoto(false))}
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: why aren'y you using <button onClick> here?

File: src/carousel/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  }
+  renderButtons(className, clickFunctionReference) {
+    const props = {
+      className: `${className} ${this.state.isSliding ? ' disabled' : ''}`,
+      onClick: clickFunctionReference
+    };
+    return <div {...props} />;
+  }
+  cloneImages(imgs) {
+    return [imgs[imgs.length - 1], ...imgs, imgs[0]];
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: one liners are cool but common sense variables are even cooler. please refactor

File: src/carousel/Carousel.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    this.currentImageIndex += changeToNext ? 1 : -1;
+    if (this.isTheLastImage(changeToNext)) {
+      setTimeout(() => {
+        this.currentImageIndex = changeToNext ? 1 : this.imagesCount;
+        this.moveSliderContainer(false);
+      }, this.transitionDuration);
+    }
+    this.moveSliderContainer();
+    this.enableSliding();
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: this is confusing. isn't the sliding enabled while the photo is changing? wouldn't it make more sense to disableButtonsWhilePhotosAreMoving?

File: src/carousel/Carousel.js (hunk 1 above)
Comment (sebikovacs) [inline src/carousel/Carousel.js]: why are you using a function here instead of html elements?

File: src/carousel/Carousel.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    this.setState({ leftPosition: -1 * this.carouselContainerWidth });
+  }
+  render() {
+    let { imgs } = this.props;
+    imgs = this.cloneImages(imgs);
+    return (
+      <div className="carousel" ref={ref => (this.carouselElement = ref)}>
+        {this.renderSliderElement(imgs)}
+        {this.renderButtons('prev', () => this.changePhoto(false))}
+        {this.renderButtons('next', () => this.changePhoto(true))}
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: use a button element with an on click handler here

File: src/carousel/Carousel.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  componentDidMount() {
+    this.carouselContainerWidth = this.carouselElement.offsetWidth;
+    this.setState({ leftPosition: -1 * this.carouselContainerWidth });
+  }
+  render() {
+    let { imgs } = this.props;
+    imgs = this.cloneImages(imgs);
+    return (
+      <div className="carousel" ref={ref => (this.carouselElement = ref)}>
+        {this.renderSliderElement(imgs)}
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: no need for a function here, you can create a var above and render the images there and then just drop everything in the return between curly braces

File: src/carousel/Carousel.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import './Carousel.css';
+
+export default class Carousel extends Component {
+  constructor(props) {
+    super(props);
+    this.currentImageIndex = 1;
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: why isn't this kept in the state?

File: src/carousel/Carousel.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    this.transitionDuration = this.props.transitionDuration;
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: why are you adding the duration as a prop to `this`?

File: src/carousel/Carousel.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    this.imagesCount = this.props.imgs.length;
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: same here

File: src/carousel/Carousel.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    this.state = { isSliding: false, isAnimating: false, leftPosition: 0 };
+    this.changePhoto = this.changePhoto.bind(this);
+  }
+  /* */
+  componentDidMount() {
+    this.carouselContainerWidth = this.carouselElement.offsetWidth;
+    this.setState({ leftPosition: -1 * this.carouselContainerWidth });
+  }
+  render() {
+    let { imgs } = this.props;
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: why let and not const

File: src/carousel/Carousel.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+      });
+    }, this.transitionDuration);
+  }
+  enableSliding() {
+    this.setState({
+      isSliding: true
+    });
+    this.disableSliding();
+  }
+  moveSliderContainer(animate = true) {
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: refactor this. there's no need to call setState at every line. do your computations and call set state at the end

File: src/carousel/Carousel.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    if (animate) {
+      this.setState({ isAnimating: animate });
+      setTimeout(() => {
+        this.setState({ isAnimating: false });
+      }, this.transitionDuration);
+    }
+    this.setState({ leftPosition: -1 * this.currentImageIndex * this.carouselContainerWidth });
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: make a variable out of the math operations

File: src/carousel/Carousel.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  }
+  isTheLastImage(leftDirection) {
+    const rightDirection = !leftDirection;
+    return (this.currentImageIndex > this.imagesCount && leftDirection) || (this.currentImageIndex < 1 && rightDirection);
```
Comment (sebikovacs) [inline src/carousel/Carousel.js]: this is very confusing. please refactor

**Pull request merged**